bigquery/tests/unit/test_client.py

```diff
@@ -3100,6 +3104,24 @@ def test_load_table_from_uri_w_client_location(self):
             method="POST", path="/projects/other-project/jobs", data=resource
         )
 
+    def test_load_table_from_uri_w_invalid_job_config(self):
+        JOB = "job_name"
+        DESTINATION = "destination_table"
+        SOURCE_URI = "http://example.com/source.csv"
+
+        creds = _make_credentials()
+        http = object()
+        job_config = object()
```

Comment:
(applies to multiple places)

It would make sense to tighten the check and also test with a  `_JobConfig` instance of an incompatible job type, e.g. `QueryJobConfig`, which is more likely the error that can be expected in practice.

Users probably pass an incorrect job config type way more often than a non-`_JobConfig` instance.